doc/app/SM_AT_DFU_commands.rst

```diff
@@ -0,0 +1,390 @@
+* The ``<addr>`` parameter is an integer indicating the address offset for the data.
+* The ``<len>`` parameter is an integer indicating the length of the data chunk to write.
+
+After the command returns ``OK``, the |SM| application enters data mode to receive exactly ``<len>`` bytes of firmware data.
+When the data has been received and written, the |SM| application sends a ``#XDFU`` unsolicited notification with the result.
+
+Example
+~~~~~~~
+
+.. code-block:: none
+
+   AT#XDFUWRITE=0,0,4096
+   OK
+   // 4096 bytes of firmware data
+   #XDFU: 0,1,0
+
+Read command
+------------
+
+The read command is not supported.
+
+Test command
+------------
+
+The test command tests the existence of the command and provides information about the type of its subparameters.
+
+Syntax
+~~~~~~
+
+.. code-block:: none
+
+   #XDFUWRITE=?
+
+Response syntax
+~~~~~~~~~~~~~~~
+
+.. code-block:: none
+
+   #XDFUWRITE: (0,1,2),<addr>,<len>
+
+Example
+~~~~~~~
+
+.. code-block:: none
+
+   AT#XDFUWRITE=?
+
+   #XDFUWRITE: (0,1,2),<addr>,<len>
+
+   OK
+
+DFU apply #XDFUAPPLY
+====================
+
+The ``#XDFUAPPLY`` command finalizes the firmware update and schedules it for activation.
+
+Set command
+-----------
+
+The set command applies the firmware update.
+
+Syntax
+~~~~~~
+
+.. code-block:: none
+
+   #XDFUAPPLY=<type>
+
+* The ``<type>`` parameter is an integer indicating the DFU image type (``0``, ``1``, or ``2``).
+
+For application and delta modem firmware, the update is scheduled and will be activated on the next reset.
+For full modem firmware, the command applies the current segment (bootloader or firmware) and may trigger a reboot.
+
+Example
+~~~~~~~
+
+.. code-block:: none
+
+   AT#XDFUAPPLY=0
+   #XDFU: 0,2,0
+   OK
+
+   AT#XRESET
+   OK
+   Ready
+
+Read command
+------------
+
+The read command is not supported.
+
+Test command
+------------
+
+The test command tests the existence of the command and provides information about the type of its subparameters.
+
+Syntax
+~~~~~~
+
+.. code-block:: none
+
+   #XDFUAPPLY=?
+
+Response syntax
+~~~~~~~~~~~~~~~
+
+.. code-block:: none
+
+   #XDFUAPPLY: (0,1,2)
+
+Example
+~~~~~~~
+
+.. code-block:: none
+
+   AT#XDFUAPPLY=?
+
+   #XDFUAPPLY: (0,1,2)
+
+   OK
+
+DFU unsolicited notification #XDFU
+==================================
+
+The |SM| application sends the ``#XDFU`` unsolicited notification to report the result of DFU write and apply operations.
+
+Unsolicited notification
+------------------------
+
+.. code-block:: none
+
+   #XDFU: <type>,<operation>,<result>
+
+* The ``<type>`` value is an integer indicating the DFU image type (``0``, ``1``, or ``2``).
+* The ``<operation>`` value is an integer indicating the operation:
+
+  * ``1`` - Data write completed
+  * ``2`` - Apply update completed
+
+* The ``<result>`` value is an integer indicating the result.
+  ``0`` indicates success, negative values indicate errors.
```

Comment:
We use both `<result>` and `<status>` for this purpose. I'd probably prefer `<status>` and it's used in a bit more places but not much.

About the possible values and especially errors. I know we return errno codes in some places. But we also have places where we put 0 or -1. The reason is that if we give errno codes, we should document what error codes can be returned and what they mean in that context. Hence, I'd suggest we just do:
`It is ``0`` for success or ``-1`` for failure.`
The reason is that the error code doesn't provide programmatic recovery based on error codes. And RTT log should then give an error code and other possible info on what's wrong.